### PR TITLE
Add font: 0xProto

### LINKED ISSUE #1379: Introduce 0xProto font
**Requirements**

- [ X ] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my request and found nothing related and/or helpful
- [ X ] I have searched the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting) for help
- [ X ] I have searched the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki) for help

**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
Please consider converting the 0xProto font, available here:

https://github.com/0xType/0xProto

**Describe alternatives you've considered**
N/A

**Additional context**
This has become my new favorite programming font for its readability and handling of ligatures, but it looks the icons and other niceties that Nerd Fonts brings.

Thanks.

### PR DESCRIPTION
**Pull request by piraces**
#### Description

Add new font 0xProto

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [x] Verified the license of any newly added font, glyph, or glyph set

#### What does this Pull Request (PR) do?

Adds the font 0xProto

#### How should this be manually tested?

N/A

#### Any background context you can provide?

I followed the entire contributing guide and tried to base this on other fonts added in successfully merged PRs.

#### What are the relevant tickets (if any)?

Closes #1379 

#### Screenshots (if appropriate or helpful)

N/A

### CODE REVIEW

**Comment by piraces** [comment]
@all-contributors please add @piraces for code

**Comment by piraces** [comment]
Hi there! First time contributing to Nerd Fonts.
I followed the entire contributing guide and tried to base this on other fonts added in successfully merged PRs.

Please let me know if I have something wrong 🙏

File: bin/scripts/lib/fonts.json [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
   "fonts": [
+    {
+      "unpatchedName": "0xProto",
+      "RFN": false,
+      "patchedName": "0xProto",
+      "folderName": "0xProto",
+      "imagePreviewFont": "0xProto",
+      "linkPreviewFont": "0xProto",
+      "caskName": "0xProto",
+      "repoRelease": true,
```
Comment (Finii) [inline bin/scripts/lib/fonts.json:11]: For all new fonts this should be `false`

File: patched-fonts/0xProto/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# 0xProto
```
Comment (Finii) [inline patched-fonts/0xProto/README.md:1]: This file and all others in `patched-fonts/` is autogenerated and should not be added manually.
* This README
* all font files

**Review by Finii — COMMENTED** [review]
This looks almost good! :-)

Adding a new font is of course always also kind of 'political'.
Is the font interesting enough?
Is there demand?
Etc.
That is of course hard for newly created fonts.

**Comment by Finii** [comment]
Forgot to write _first of all_:

Thank you for the PR :-)

File: patched-fonts/0xProto/README.md (hunk 2 above)
Comment (piraces) [reply patched-fonts/0xProto/README.md]: Understood! Removed all content from `patched-fonts/`

File: bin/scripts/lib/fonts.json (hunk 1 above)
Comment (piraces) [reply bin/scripts/lib/fonts.json]: Changed 👍

**Comment by piraces** [comment]
Hi @Finii, first of all thank you for all the explication!
It´s awesome to contribute to a repo and have this kind of warm welcoming ☺️ 

I think I understood the process now. I have made the suggested changes in overwriting the same commit and "force pushing" (let me know if you are not ok with "force pushes").

---

**Off-topic (kind of):** thinking about the [contributing.md, adding a new font guide section](https://github.com/ryanoasis/nerd-fonts/blob/master/contributing.md#steps-for-adding-a-new-font-or-removing-an-existing-font), I was thinking it may be useful to change it with this "new" steps (I can try to work on that too). Should we track this proposal in a new issue or discussion? Thanks! 👍

File: bin/scripts/lib/fonts.json [hunk 3]
```diff
@@ -1,5 +1,16 @@
 {
   "fonts": [
+    {
+      "unpatchedName": "0xProto",
+      "RFN": false,
+      "patchedName": "0xProto",
+      "folderName": "0xProto",
+      "imagePreviewFont": "0xProto",
+      "linkPreviewFont": "0xProto",
```
Comment (Finii) [inline bin/scripts/lib/fonts.json:9]: I'm not sure but is this not:

```suggestion
      "linkPreviewFont": "oxproto",
```

Interestingly the zero is replaced by letter o?!

image

**Comment by Finii** [comment]
I wrote something on adding fonts [here](https://github.com/ryanoasis/nerd-fonts/issues/1335#issuecomment-1736639242): 

image


> and "force pushing" (let me know if you are not ok with "force pushes")

Force pushes are absolutely ok in PRs (and better than having 10 commits that fix fixes ;)

> **Off-topic (kind of):** thinking about the [contributing.md, adding a new font guide section](https://github.com/ryanoasis/nerd-fonts/blob/master/contributing.md#steps-for-adding-a-new-font-or-removing-an-existing-font), I was thinking it may be useful to change it with this "new" steps (I can try to work on that too). Should we track this proposal in a new issue or discussion?

Ah, that is rather old and I completely forgot that section.
I'm not so sure with add-a-font PR; in principle we would need to fix our repo setup first which we struggle since some years and there is no solution in sight. And adding a new font is - in contrast to when that paragraph had been added - rather trivial. Hmm.
The problem is that this repo grew exponentially with each added font, this is not sustainable.

I still have no solutions for the problems, and the `inRepoRelease = false` is just the first ministep into the right direction but not sufficient I fear.

What I wanted to say: I would rather _not_ promote add-a-font PRs.
I really would love fix-a-bug PRs or something ;-D

File: bin/scripts/lib/fonts.json (hunk 3 above)
Comment (piraces) [reply bin/scripts/lib/fonts.json:9]: Kind of strange... yes...

**Comment by piraces** [comment]
I just have made the changes.

> What I wanted to say: I would rather not promote add-a-font PRs.
I really would love fix-a-bug PRs or something ;-D

I completely understand your point, if I have more time I will work on bugs or something related better than new fonts 😄

**Comment by adrianvalenz** [comment]
> This looks almost good! :-)
> 
> Adding a new font is of course always also kind of 'political'. Is the font interesting enough? Is there demand? Etc. That is of course hard for newly created fonts.

Can I just comment to support this PR...0xProto is an amazing font to work with.

**Comment by Finii** [comment]
I mark this as ready now (i.e. I will update the font just on release, if need be).
Thank you for the work :+1: 

Ah, I will quickly add a commit that unifies the readme style.

**Comment by piraces** [comment]
Thank you for all @Finii !! 👍

**Comment by neotpravlennoye** [comment]
0xProto is my favorite code font and I'm glad it's joining nerd font family.
Thanks to all of you for your efforts 👍

**Pull request merged**